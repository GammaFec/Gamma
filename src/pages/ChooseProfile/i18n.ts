import i18next from "i18next";

import { LanguageList } from "./../../i18next/config";

i18next.addResources(LanguageList.SPANISH, "ChooseProfile", {
    title: "Elige tu perfil deseado",
    description:
        "En PetMatch nos importas mucho, por eso selecciona que papel deseas tener en la aplicación para brindarte una experiencia única."
});

i18next.addResources(LanguageList.ENGLISH, "ChooseProfile", {
    title: "Choose your desired profile",
    description:
        "At PetMatch we care a lot about you, so select what role you want to have in the application to give you a unique experience."
});
