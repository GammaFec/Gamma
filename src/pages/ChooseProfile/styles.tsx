import styled from "styled-components";
import { fontSizes, breakpoints } from "../../common/styles/variables";
import { ReactComponent as BlueDog } from "../../assets/img/blue_dog.svg";

export const StyledBlueDog = styled(BlueDog)`
    position: fixed;
    bottom: 0;
    right: 20px;
`;

export const StyledMain = styled.main`
    align-items: center;
    display: flex;
    justify-content: center;
    min-height: 100vh;
    width: 100vw;
`;

export const StyledContainer = styled.div`
    max-width: 90%;
    text-align: center;
`;

export const StyledTitle = styled.h1`
    font-size: ${fontSizes.font21};
    line-height: 31.5px;
    margin-bottom: 20px;

    @media (min-width: ${breakpoints.tablet}) {
        font-size: ${fontSizes.font30};
    }
`;

export const StyledText = styled.p`
    font-size: ${fontSizes.font12};
    line-height: 18px;
    margin-bottom: 45px;

    @media (min-width: ${breakpoints.tablet}) {
        font-size: ${fontSizes.font16};
    }
`;
