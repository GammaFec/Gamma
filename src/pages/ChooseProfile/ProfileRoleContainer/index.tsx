import React, { useState } from "react";
import ProfileRole from "./ProfileRole";
import { StyledButton, StyledProfileRoleContainer } from "./styles";
import rolesData from "./rolesData.json";
import { IRoleData, ISelectedRol } from "./types";

const ProfileRoleContainer: React.FC = (): JSX.Element => {
    const Data: IRoleData[] = rolesData;
    const [Selected, setSelected] = useState<ISelectedRol | null>(null);
    const [ButtonDisabled, setButtonDisabled] = useState(true);

    const handleClick = (id: string, path: string): void => {
        if (id === Selected?.id) {
            setSelected(null);
            setButtonDisabled(true);
        } else {
            setSelected({ id, path });
            setButtonDisabled(false);
        }
    };

    return (
        <>
            <StyledProfileRoleContainer>
                {Data.map(({ id, path, variant, name }, i) => (
                    <ProfileRole
                        handleClick={(): void => handleClick(id, path)}
                        key={i}
                        selected={Selected?.id === id}
                        variant={variant}>
                        <p>{name}</p>
                    </ProfileRole>
                ))}
            </StyledProfileRoleContainer>
            <StyledButton
                className={`${ButtonDisabled && "disabled"}`}
                disabled={ButtonDisabled}
                variant="primary">
                Continuar
            </StyledButton>
        </>
    );
};

export default ProfileRoleContainer;
