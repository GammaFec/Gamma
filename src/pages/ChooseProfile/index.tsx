import React from "react";
import { useTranslation } from "react-i18next";
import ProfileRoleContainer from "./ProfileRoleContainer";
import { StyledContainer, StyledMain, StyledText, StyledTitle } from "./styles";
import rolesData from "./ProfileRoleContainer/rolesData.json";

const ChooseProfile: React.FC = (): JSX.Element => {
    const { t } = useTranslation("ChooseProfile");
    return (
        <StyledMain>
            <StyledContainer>
                <StyledTitle>{t("title")}</StyledTitle>
                <StyledText>{t("description")}</StyledText>
                <ProfileRoleContainer Data={rolesData} handleSubmit={(): void => alert("submit")} />
            </StyledContainer>
        </StyledMain>
    );
};

export default ChooseProfile;
