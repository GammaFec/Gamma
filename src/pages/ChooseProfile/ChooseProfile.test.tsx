import React from "react";
import ChooseProfile from "./";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import "jest-styled-components";

jest.mock("react-i18next", () => ({
    useTranslation: (): { t: (key: string) => string } => ({ t: (key: string): string => key })
}));

describe("<ChooseProfile />", () => {
    test("Render", () => {
        const { asFragment } = render(<ChooseProfile />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Render all childrens", () => {
        const { getByText } = render(<ChooseProfile />);

        getByText("title");
        getByText("description");
        getByText("continue");
    });
});
