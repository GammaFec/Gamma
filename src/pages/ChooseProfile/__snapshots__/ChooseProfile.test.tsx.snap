// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChooseProfile /> Render 1`] = `
<DocumentFragment>
  .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 148px;
  height: 153px;
  background-color: #FF7B28;
  border-radius: 7px;
  box-shadow: 1px 5px 20px -4px rgba(0,0,0,0.67);
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  -webkit-transition: -webkit-transform 0.1s linear;
  -webkit-transition: transform 0.1s linear;
  transition: transform 0.1s linear;
  cursor: pointer;
  font-weight: lighter;
}

.c5:hover {
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c5 * {
  margin: 0;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  width: 100%;
  background-color: rgba(0,0,0,0.2);
  color: #FFFFFF;
  font-size: 12px;
  font-weight: bolder;
}

.c5.selected {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c5:not(.selected) {
  opacity: 0.9;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 148px;
  height: 153px;
  background-color: #00A59E;
  border-radius: 7px;
  box-shadow: 1px 5px 20px -4px rgba(0,0,0,0.67);
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  -webkit-transition: -webkit-transform 0.1s linear;
  -webkit-transition: transform 0.1s linear;
  transition: transform 0.1s linear;
  cursor: pointer;
  font-weight: lighter;
}

.c6:hover {
  -webkit-transform: rotate(5deg);
  -ms-transform: rotate(5deg);
  transform: rotate(5deg);
}

.c6 * {
  margin: 0;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  width: 100%;
  background-color: rgba(0,0,0,0.2);
  color: #FFFFFF;
  font-size: 12px;
  font-weight: bolder;
}

.c6.selected {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c6:not(.selected) {
  opacity: 0.9;
}

.c7 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 16px;
  font-family: 'Poppins','sans-serif';
  border-radius: 10px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  line-height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  padding: 0.7rem 3rem;
  margin: 0.2rem;
  min-width: 243px;
  height: 48px;
  cursor: pointer;
  background-color: #FF7B28;
  border: #FF7B28 solid 2px;
  color: #FFFFFF;
}

.c7.disabled {
  cursor: not-allowed;
  opacity: 0.6;
}

.c7:hover {
  opacity: 0.6;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  list-style: none;
  margin-bottom: 30px;
  margin-top: 1rem;
  padding: 0;
  width: 100%;
}

.c8 {
  font-size: 18px;
  font-weight: lighter;
  margin: auto;
  max-width: 600px;
  width: 100%;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 100vh;
  width: 100vw;
}

.c1 {
  max-width: 90%;
  text-align: center;
}

.c2 {
  font-size: 21px;
  line-height: 31.5px;
  margin-bottom: 20px;
}

.c3 {
  font-size: 12px;
  line-height: 18px;
  margin-bottom: 45px;
}

@media (min-width:720px) {
  .c2 {
    font-size: 30px;
  }
}

@media (min-width:720px) {
  .c3 {
    font-size: 16px;
  }
}

<main
    class="c0"
  >
    <div
      class="c1"
    >
      <h1
        class="c2"
      >
        title
      </h1>
      <p
        class="c3"
      >
        description
      </p>
      <ul
        class="c4"
      >
        <div
          class="c5 false"
          color="primary"
        >
          <p>
            Adoptante
          </p>
        </div>
        <div
          class="c6 false"
          color="secondary"
        >
          <p>
            Albergue
          </p>
        </div>
      </ul>
      <button
        class="c7 c8 disabled"
        disabled=""
      >
        continue
      </button>
    </div>
  </main>
</DocumentFragment>
`;
